tests/testthat/test-survival-tune-autoplot.R

```diff
@@ -0,0 +1,1038 @@
+
+  skip_if_not_installed("rpart")
+  skip_if_not_installed("prodlim")
+
+  # ------------------------------------------------------------------------------
+
+  set.seed(1)
+  sim_dat <- prodlim::SimSurv(1000) %>%
+    mutate(event_time = Surv(time, event)) %>%
+    select(event_time, X1, X2)
+
+  set.seed(2)
+  split <- initial_validation_split(sim_dat)
+  sim_tr <- training(split)
+  sim_te <- testing(split)
+  sim_rs <- validation_set(split)
+
+  time_points <- c(10, 1, 5, 15)
+
+  mod_spec <-
+    decision_tree(cost_complexity = tune(), min_n = tune()) %>%
+    set_engine("rpart") %>%
+    set_mode("censored regression")
+
+  gctrl <- control_grid(save_pred = TRUE)
+  bctrl <- control_bayes(save_pred = TRUE)
+
+  # Bayes with dynamic metric --------------------------------------------------
+
+  dyn_mtrc  <- metric_set(brier_survival)
+
+  set.seed(2193)
+  grid_dynamic_res <-
+    mod_spec %>%
+    tune_grid(
+      event_time ~ X1 + X2,
+      resamples = sim_rs,
+      grid = 10,
+      metrics = dyn_mtrc,
+      eval_time = time_points,
+      control = gctrl
+    )
+
+  expect_snapshot({
+    set.seed(2193)
+    bayes_dynamic_res <-
+      mod_spec %>%
+      tune_bayes(
+        event_time ~ X1 + X2,
+        resamples = sim_rs,
+        iter = 2,
+        metrics = dyn_mtrc,
+        eval_time = time_points,
+        control = bctrl,
+        initial = grid_dynamic_res
+      )
+  })
+
+  # ------------------------------------------------------------------------------
+
+  # default to first time
+  dyn_grid <- autoplot(grid_dynamic_res)
+
+  exp_data_ptype <-
+    tibble::tibble(
+      mean = numeric(0),
+      `# resamples` = integer(0),
+      .metric = character(0),
+      name = character(0),
+      value = numeric(0)
+    )
+
+  expect_equal(exp_data_ptype, dyn_grid$data[0,])
+  expect_equal(
+    unique(dyn_grid$data$name),
+    c("Cost-Complexity Parameter (log-10)", "Minimal Node Size")
+  )
+  expect_equal(
+    rlang::expr_text(dyn_grid$mapping$x),
+    "~value"
+  )
+  expect_equal(
+    rlang::expr_text(dyn_grid$mapping$y),
+    "~mean"
+  )
+  expect_equal(
+    rlang::expr_text(dyn_grid$facet$params$facets$name),
+    "~name"
+  )
+  expect_equal(
+    map(dyn_grid$labels, as.character),
+    list(y = "brier_survival @10", x = "")
+  )
+
+  # ------------------------------------------------------------------------------
+
+  # multiple times
+  dyn_mult_grid <- autoplot(grid_dynamic_res, eval_time = c(1, 10))
+
+  exp_data_ptype <-
+    tibble::tibble(
+      mean = numeric(0),
+      `# resamples` = integer(0),
+      .metric = character(0),
+      name = character(0),
+      value = numeric(0)
+    )
+
+  expect_equal(exp_data_ptype, dyn_mult_grid$data[0,])
+  expect_equal(
+    unique(dyn_mult_grid$data$name),
+    c("Cost-Complexity Parameter (log-10)", "Minimal Node Size")
+  )
+  expect_equal(
+    unique(dyn_mult_grid$data$.metric),
+    c("brier_survival @ 1", "brier_survival @10")
```

Comment:
I don't like this 😄  why is there no space after `@` in one but not the other?

Reply:
I use `format()` on the times so they order correctly in facets: 

```
> sort(paste(1:10))
 [1] "1"  "10" "2"  "3"  "4"  "5"  "6"  "7"  "8"  "9" 
> sort(format(1:10))
 [1] " 1" " 2" " 3" " 4" " 5" " 6" " 7" " 8" " 9" "10"
```

Reply:
I could do the recipes naming thing and add the leading zero: 

```
> sort(gsub(" ", "0", format(1:10)))
 [1] "01" "02" "03" "04" "05" "06" "07" "08" "09" "10"
```

Reply:
It looks a lot weirder with decimal numbers though.

Reply:
that would be fine i guess 😄

Reply:
oh no. don't sweat too much. I can live with the current spacing